byron/chain/executable-spec/test/Cardano/Spec/Chain/STS/Properties.hs

```diff
@@ -78,3 +85,110 @@ signersListIsBoundedByK = property $ do
         signersListIsBoundedByKInState :: MonadTest m => BlockCount -> State CHAIN -> m ()
         signersListIsBoundedByKInState (BlockCount k') (_sLast, sgs, _h, _utxoSt, _ds, _us) =
           assert $ length sgs <= fromIntegral k'
+
+
+relevantCasesAreCovered :: Property
+relevantCasesAreCovered = withTests 400 $ property $ do
+  tr <- forAll $ traceSigGen (Desired 250) (sigGenChain GenDelegation NoGenUTxO NoGenUpdate)
+  let certs = traceDCerts tr
+
+  -- for at least 1% of traces...
+  cover 1
+        "there are more certificates than blocks"
+        (traceLength tr <= length certs)
+
+  -- for at least 10% of traces...
+  cover 10
+        "at most 75% of blocks have no certificates"
+        (emptyDelegationPayloadRatio (traceDCertsByBlock tr) <= 0.75)
+
+  -- for at least 25% of traces...
+  cover 25
+        "at least 25% of delegates will delegate to this epoch"
+        (0.25 <= thisEpochDelegationsRatio (epochDelegationEpoch tr))
+
+  -- for at least 60% of traces...
+  cover 60
+        "at least 50% of delegations will delegate to the next epoch"
+        (0.5 <= nextEpochDelegationsRatio (epochDelegationEpoch tr))
+
+  -- for at least 20% of traces...
+  cover 20
```

Comment:
In the test I ran this results in 20% of coverage, we might want to either change this to 10% or change the threshold below to `.25`.